docs/multiple-canaries-guide.md

```diff
@@ -0,0 +1,112 @@
+
+The local canary itself is very simple, but when it comes to the relationship between them we need to be careful. So we will use an evolutionary way to illustrate its points.
+
+As the basic example at (1), we have 3 services to represent backend services of an order takeaway app, and primary traffic means all traffic except canary traffic:
+
+![image](imgs/multiple-canaries-guide-01.png)
+
+Then at (2), `team-delivery` deployed a local canary `delivery-beijing` to test a new feature for traffic from Beijing.
+
+![image](imgs/multiple-canaries-guide-02.png)
+
+And we go to (3), another team `team-restaurant` deployed another local canary `restaurant-beijing` to test another new feature. So if the two canaries handled some or all common traffic, the clients might get unexpected results. For example, `restaurant-beijing` returned a cook duration but `delivery-beijing` returned a delivery duration, where the sum of two separated durations is not consistent with the original total duration. This kind of confusing situation isn’t absolutely what we want to appear.
+
+![image](imgs/multiple-canaries-guide-03.png)
+
+So the situation illustrated (4) is what we expect in normal scenarios. We need to explicitly split Beijing traffic into two parts respectively going through two different canaries. There are different solutions in different environments, we will demonstrate one later.
+
+![image](imgs/multiple-canaries-guide-04.png)
+
+As the evolution shows, we can tell it’s unsafe that local canaries share some part of traffic. In other words technically: Local canaries do not call each other.
+
+## Global Canary
+
+Based on local canary, the term global canary is pretty clear, it is for the feature that needs multiple services to respectively deploy one release to support one canary. Along with the local canary example, we evolve it with global canary:
+
+![image](imgs/multiple-canaries-guide-05.png)
+
+The principles of global canary evolved from local canary is almost the same: It can’t share traffic with other local or global canaries. But as a global canary, it needs one more principle: It needs to call the same global release of another service if there is, otherwise the primary release can be just the choice. If the traffic choice violates the principles, it can’t get the whole part of the feature, or even it could get unsafe behavior.
+
+When we reach here, we can tell clearly: **Local canary is just a special case of global canary**. So we can conclude 2 core principles here for multiple canary deployments:
+
+1. The traffic rules of choosing the canary are explicitly exclusive.
+2. The complete chain of a request only goes to zero or only one canary.
+
```

Comment:
It's important to emphasize that the canary policy will never schedule the coloring traffics to normal services' instances, as it will bring huge business or technical risk to users.

Reply:
Updated.